### DISCUSSION TITLE
I know a .air file can make a standalone .exe which installs air... Can I take a .air file and make a native capture air .exe?

### DISCUSSION THREAD

**Discussion by goodnewsjimdotcom** [airsdk/Adobe-Runtime-Support#2462] [Q&A]
Hello,

The only way I know how to make a native capture air .exe is to have a .swf.

Background why I want to make a native capture without .swf and just .air:

My company Tangerine Pop has two Flash games which are incredibly deep, probably 300,000 lines of hand written code, one file with 50,000 lines alone!  It's definitely one of the more flash games ever, we got 4 million plays on Shockwave, Dungeon Run a tribute to Gauntlet 2.  The problem is with a code base of 300,000 lines, converting from .apk and air down to .swf is difficult.  I'm stuck on trying to find out how to load .swf files and parse the image cutters by downgrading backwards from Starling.  I wish I could just take a .air file and make a native .air installer .exe.  Steam/Valve will not accept self installing air by their policy, but they do allow native capture .exe to install.  Is the feature to take a .air file and become a native installer .exe there, or can you guys do it later?  I can only make a standalone .exe that installs air with adt at the moment.

If you wanna see Dungeon Run:

https://www.youtube.com/watch?v=-77GCNXPIqc

,

**Comment by ajwfrost** [comment]
This should be possible with the `-target bundle` option in ADT? `-target native` will make the .exe native installer as you mention, which doesn't include the runtime, but if you use `bundle` it will create the necessary application folder structure including the .exe file that launches the game, the distributable AIR runtime, and all the game assets...

**Reply by sorrex** [reply]
I have also pretty big application (for WIndows), but only about 200.000 lines of hand written code :)

I use -target bundle to make application folder structure including the .exe file as [ajwfrost](https://github.com/ajwfrost) recommends.
Than I use Enigma Protector to encrypt and package application into one executable file.
In reality I have split the application into several packages due to update process of application - some stuff do not need updates while others are updated more frequently.
Im surprised than nobody mentions this process here. Have anybody else using this workflow? Or any other suggestions how to make windows distribution, without exposing main swf?

**Reply by FliplineStudios** [reply]
@sorrex Thanks for sharing about Enigma Protector!  I'd love to hear what others are doing to obscure the main SWF in Windows releases too.

I was also surprised that the .exe which `-target bundle` creates doesn't seem to require the same SWF that you originally packaged with -- you can easily edit the application XML in the META-INF/AIR folder after it's been packaged, and can swap out your app's main SWF with any old SWF after packaging and end users can just use it like a glorified Flash Player at that point.

**Reply by sorrex** [reply]
Yes, you are correct - main .exe is just loader and is not tied to actual swf. And that is a feature which I use to update application. The main package (simplified) contains this loader.exe and original Adobe stuff (dlls, licence etc) and my swf is packed into another packet. Then, when I need to update application, I just update this very small packet with actual swf without need to upload the other stuff.

Working with packagers like Enigma is really easy and you will find it fast. You can use free Enigma Virtual Box for the same purpose. But there are some good reasons why I use Enigma Protector.

**Comment by pol2095** [comment]
You can create a [msi](https://web.archive.org/web/20120328014428if_/http://www.adobe.com/devnet/air/articles/customize-setup-for-AIR-app-with-captive-runtime.html) too and after a msix for Windows Store.
